BlazarData/src/main/java/com/hubspot/blazar/data/dao/StateDao.java

```diff
@@ -40,39 +41,52 @@
       "WHERE gitInfo.id = :branchId")
   Optional<RepositoryState> getRepositoryState(@Bind("branchId") int branchId);
 
-  @SingleValueResult
+  /**
+   * @param branchId The branch we are fetching module state for
+   * @return Optional(ModuleState) (partial)
+   *
+   * Complete state of a module is a very complex thing spanning many different builds
+   * - lastSuccessful
+   * - lastNonSkipped
+   * - lastCompleted
+   * - pending
+   * - inProgress
+   *
+   * To enhance performance, #getPartialModuleStatesForBranch retrieves the following data for each module on a branch:
+   * - the module
+   * - lastCompleted
+   * - pending
+   * - inProgress
+   *
+   * It returns these in the form of a partially filled ModuleState object, A set of ModuleBuildInfo for
+   * lastSuccessful and lastNonSkipped can be fetched with #getLastSuccessfulAndNonSkippedBuildInfos.
+   */
   @SqlQuery("" +
       "SELECT * " +
-      "  FROM modules AS module " +
+      "  FROM branches " +
+      "     LEFT OUTER JOIN modules as module on (branches.id = module.branchId) " +
       // last build
       "     LEFT OUTER JOIN module_builds AS lastModuleBuild ON (module.lastBuildId = lastModuleBuild.id) " +
-      "     LEFT OUTER JOIN repo_builds AS lastRepoBuild ON (lastModuleBuild.repoBuildId = lastRepoBuild.id) " +
+      "     LEFT OUTER JOIN repo_builds AS lastBranchBuild ON (lastModuleBuild.repoBuildId = lastRepoBuild.id) " +
```

Comment:
This query is wrong `repo_builds AS lastBranchBuild` but you have `lastRepoBuild.id`